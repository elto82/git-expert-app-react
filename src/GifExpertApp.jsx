import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GitGrid } from "./components/GitGrid";

export const GifExpertApp = () => {

  const [ categories, setCategories ] = useState([ 'One Punch' ]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    // console.log(newCategory);
    setCategories([ newCategory, ...categories ]);

  };

  return (
    <>
      {/* titulo */}
      <h1>GifExpertApp</h1>
      {/* input */}

      <AddCategory
        // setCategories={setCategories}
        onNewCategory={onAddCategory}
      />
      {/* listado de Gif */}
      {categories.map(category => (
        <GitGrid key={category} category={category} />
      ))}

    </>
  );
};
