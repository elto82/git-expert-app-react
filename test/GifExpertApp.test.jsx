import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Test component <GifExpertApp', () => {
  test('should show the title', () => {
    render(<GifExpertApp />);
    const { container } = render(<GifExpertApp />);
    const h1 = container.querySelector('h1');
    expect(h1.innerHTML).toContain('GifExpertApp');
    expect(h1).toMatchSnapshot();
    // screen.debug();
  });

  // Tests that the component renders without throwing an error
  test('should render without errors', () => {
    render(<GifExpertApp />);
    expect(true).toBeTruthy();
  });

});
