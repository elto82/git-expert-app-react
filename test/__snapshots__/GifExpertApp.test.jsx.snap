// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifExpertApp should show the title 1`] = `
<h1>
  GifExpertApp
</h1>
`;
